server/service/src/processors/README.md

```diff
@@ -0,0 +1,33 @@
+# Processors
+
+It is desirable to perform some operations consecutively as they may cause undesirable and unexpected side effects. These operations should be `triggerable` in services, wherever `ServiceContext` and `ServiceProvider` are available and should be `triggerable` on individual basis.
+
+We can achieve this creating channels and spawning a task that listens to those channels. The senders for the channel can be put into Atomic Reference Counter (Arc) and passed along as a context to a service.
+
+## Implementation
+
+Channels are started via `Processors::init()`, and service provider takes the `Sender` part of the `Processors::init()` result tuple (`Sender` is already `Arc`ed). `Processors::spawn()` will start a task and return the `handle`. The task will wait on a message sent to trigger a particular processor and will invoke that processor.
```

Comment:
:nit
```suggestion
Channels are started via `Processors::init()`, and service provider takes the `Sender` part of the `Processors::init()` result tuple (`Sender` is already `Arc'ed`). `Processors::spawn()` will start a task and return the `handle`. The task will wait on a message sent to trigger a particular processor and will invoke that processor.
```

Reply:
Since the data type is `Arc` i wanted to keep the emphasis on the `Arc`